src/platform/magento2/order/orderProcessor.js

```diff
@@ -0,0 +1,302 @@
+ * @returns {Promise<*>}
+ */
+async function assignAddresses ({ orderData, job, done }) {
+  const {
+    cart_id: cartId,
+    addressInformation: {
+      billingAddress: billingAddr,
+      shippingAddress: shippingAddr
+    }
+  } = orderData;
+  const api = apiUtilities.getApi(orderData.store_code);
+
+  try {
+    var countryList = await api.directory.countries();
+  } catch (error) {
+    if (job) job.attempts(3).backoff({ delay: 30 * 1000, type: 'fixed' }).save()
+    return done(orderUtilities.composeError('Error while syncing country list.', error));
+  }
+
+  var mappedBillingRegion = countryMapper.mapCountryRegion(
+    countryList,
+    billingAddr.country_id,
+    billingAddr.region_code ? billingAddr.region_code : billingAddr.region
+  );
+
+  var mappedShippingRegion;
+
+  if (typeof shippingAddr !== 'undefined' && shippingAddr !== null) {
+    mappedShippingRegion = countryMapper.mapCountryRegion(
+      countryList,
+      shippingAddr.country_id,
+      shippingAddr.region_code ? shippingAddr.region_code : shippingAddr.region
+    );
+  }
+
+  const billingAddressInfo = orderUtilities.getBillingAddress(billingAddr, mappedBillingRegion);
+  const shippingAddressInfo = orderUtilities.getShippingAddressInformation(
+    billingAddr,
+    mappedBillingRegion,
+    orderData,
+    mappedShippingRegion
+  );
+
+  const THREAD_ID = orderUtilities.getThreadId(job);
+  logger.info(THREAD_ID + '< Billing info', billingAddressInfo);
+
+  try {
+    const billingAddress = await api.cart.billingAddress(
+      null,
+      cartId,
+      billingAddressInfo,
+      orderUtilities.isThisAuthOrder(orderData)
+    );
+    logger.info(THREAD_ID + '< Billing address assigned', billingAddress);
+  } catch (err) {
+    logger.error('Error while adding billing address', err)
+    if (job) job.attempts(3).backoff({ delay: 60 * 1000, type: 'fixed' }).save()
+    return done(orderUtilities.composeError('Error while adding billing address.', err));
+  }
+
+  logger.info(THREAD_ID + '< Shipping info', shippingAddressInfo);
+
+  try {
+    const shippingInformation = await api.cart.shippingInformation(
+      null,
+      cartId,
+      shippingAddressInfo,
+      orderUtilities.isThisAuthOrder(orderData)
+    );
+    logger.info(THREAD_ID + '< Shipping address assigned', shippingInformation);
+    updateStep(job);
+    return shippingInformation;
+  } catch (err) {
+    logger.error('Error while adding shipping address', err)
+    if (job) job.attempts(3).backoff({ delay: 60 * 1000, type: 'fixed' }).save()
+    return done(orderUtilities.composeError('Error while adding shipping address.', err));
+  }
+}
+
+/**
+ * Verify if json order data is valid
+ * @param orderData
+ * @param job
+ * @param done
+ * @returns {boolean}
+ */
+function validateOrder ({ orderData, job, done }) {
+  const THREAD_ID = orderUtilities.getThreadId(job);
+
+  if (!orderUtilities.validateOrder(orderData)) {
+    const orderErrors = orderUtilities.getOrderErrors(orderData);
```

Comment:
You are calling in these 2 functions (line 274 & 275) `validate` method twice while it is needed only once. Please refactor it to validate only once